### ISSUE TITLE
Investigate dev-branch changes and steam workshop

### ISSUE THREAD

**Issue by JoSchaap** [JoSchaap/GoT_Wasteland_V2.Stratis#106]
Lots of new info came out during my short hiatus.
With game going live on the 12th of.sept and lots of new changes to the dev branch i need to read it all and see what's new

**Comment by AgentRev** [comment]
I still believe we should merge together to adapt Wasteland to Altis, with the help of anyone who might want to join, probably like HatchetHarry and MercyfulFate.

We'd keep it branded as GoT, but have a shared repo where those who are part of the team can edit the main code. I was also thinking about picking up Fussion like Harry did, since it's a nice addition and its development was abandoned.

I'd probably add more settings so people who don't want stuff like survival or general stores can turn them off, but the idea is to have a shared codebase instead of working separately, which can be redundant at times.

I think it would be beneficial to everyone in the long run. On a side note, I'm working on a proper selling system for gunstores.

**Comment by MercyfulFate** [comment]
I'm keen for that. I'm pretty flat out at the moment... but once things calm down with uni and work, I'll be able to put a decent amount of work into it.

**Comment by JoSchaap** [comment]
@AgentRev you definitely have a point. there will be some debate on some of the features (such as medkits and the fussion stuff)  but in the end it will result in a better mission. for now i was merely concentrating on getting the code descent, so that incase altis launches, all we have to worry about, is placing new markers and waypoints.

but a coop seems the most logical choice. mercry's inventory idea has some good potential and you are just more on top of creating descent/optimized code instead of just fuctional

imho it's a nobrainer, so yes i agree :) 

the base capture and takeover town stuff i had in my mind for a while now, my idea was to utilize the 'world-missions' for these

this would mean you'd have
- world missions every 20/30 minutes (Capture area, clear AI from town, takeover base)
- main missions every 15 minutes ( high value missions )
- side missions every 10 minutes (low/medium value missions)

also the old antihack should no longer be needed when we got proper battleye filters (saves resources)

but first things first. who would agree in a joint venture on GoT Wasteland Altis?

me
agentrev
mercyfullfate
..
..
..
..

:)

**Comment by JoSchaap** [comment]
Does anyone know who's behind the KOS adaption? :)

they also added some fresh idea's :)

**Comment by AgentRev** [comment]
Very well, this is going to be awesome. As we're gonna work simultaneously on the repository, we'll have to establish simple guidelines to avoid potential screw-ups; I learned that the hard way.

I think world mission should probably be called mercenary or bounty missions, and have :moneybag: rewards associated with them, like KoS missions.

Speaking of KoS, the main developer is <i>His_Shadow</i> : http://manumitthemind.com/kos/forum/index.php?topic=3#msg11

For the anti-hack, we'll need to keep stuff related to `BIS_fnc_MP` and `TPG_fnc_MP` because BattlEye lacks proper filtering in that area, but the rest can be scrapped.

I hope BIS will put Altis on DEV before the official release :smiley:

**Comment by JoSchaap** [comment]
rule #1 no one touches the #master branch it's for releases only, always work in a seperate branch :)

and altis will come to dev in a week or so :)

**Comment by MercyfulFate** [comment]
The way we (my work team, 30+ people) use git we have 2 main branches (master and development) and we each have our own feature branches.
master is the latest release.
development is code that is tested and ready for next release.
feature branches is where the development is done with a few rules:
- branch directly from development branch (no sub-branching to avoid issues)
- WIP for ONE "feature/fix" only.
- are re-based as soon as new work is committed to development (to catch errors early, integrate fixes, etc).
- are compressed to a single commit when committing to development (to make it easier for everyone else to rebase).

Its kind of up to everyone, but this way we don't have to commit so often, and it avoids a lot of issues merging in  multiple branches into master for a "release".

**Comment by JoSchaap** [comment]
@mercufullfate great ideas!

I've only discovered githubs release section today _shamesmiley_ but even that works great 
example: https://github.com/JoSchaap/GoT_Wasteland_V2.Stratis/releases

As far as KOS concerns i hope i found the right person: http://forums.bistudio.com/member.php?114613-His_Shadow (PM sent)

**Comment by MercyfulFate** [comment]
on a seperate note, JoShcapp why do you prefix your branches with a # (eg #master)

**Comment by AgentRev** [comment]
Yeah, for some reason this confuses GitHub and prevents branches from being downloaded as zip files, causing an error 500 when trying to do so

**Comment by JoSchaap** [comment]
i twitter too much i guess.. :) ill stop doing so

**Comment by MercyfulFate** [comment]
not only that but in your commits, you can reference github branches by using # (https://github.com/blog/831-issues-2-0-the-next-generation => Commits + issues halfway down the page)

oh and we have to get a better way of communicating than github

**Comment by AgentRev** [comment]
> WIP for ONE "feature/fix" only.

You mean per developer?

**Comment by MercyfulFate** [comment]
No I mean each feature branch focus on one thing only. multiple developers can work on the same feature branch.

**Comment by AgentRev** [comment]
Ah, okay. Sounds good.

**Comment by AgentRev** [comment]
> oh and we have to get a better way of communicating than github

I think <a href="http://hall.com/">Hall</a> could be a good option. It's free for personal use, and it has <a href="http://hall.com/docs/integrations/github/">GitHub integration</a>.

**Comment by MercyfulFate** [comment]
I'm happy to use hall. up to you guys.

**Comment by JoSchaap** [comment]
hall it is :)

![halltest](https://f.cloud.github.com/assets/2301588/967677/4615e7cc-0583-11e3-8704-3c87e1319173.jpg)

**Comment by MercyfulFate** [comment]
Request Send on Hall.

**Comment by JoSchaap** [comment]
steam workshop test mission uploaded :)

http://steamcommunity.com/sharedfiles/filedetails/?id=169806722

or just search the workshop for wasteland

**Comment by JoSchaap** [comment]
There are 3 Classes of NVGoggles now (DEVbranch)

"NVGoggles"
"NVGoggles_INDEP"
"NVGoggles_OPFOR"

Ill wait till they rename the top one to _BLUFOR  :/

so IND and OPFOR will spawn with NVG again since removeitem only removes the top one (blufor)

**Comment by AgentRev** [comment]
```
{
    if (["NVGoggles", _x] call fn_findString == 0) exitWith
    {
        _player unassignItem _x;
        _player removeItem _x;
    };
} forEach assignedItems _player;
```

**Comment by JoSchaap** [comment]
in the GoT version it isnt such a problem, GoT does not remove NVG from players

**Comment by AgentRev** [comment]
Then why the "so IND and OPFOR will spawn with NVG again"?

**Comment by JoSchaap** [comment]
for your version :)

i dont have nvg in the loot/stores i just let players spawn with them
there was too much swearing about how hard they where to get on dark servers...

**Comment by AgentRev** [comment]
Oh! Ok, thanks.

Yeah, TPG servers are always reset when they hit nighttime, although I do like playing during the night without NVGs. Either with a full moon, or max gamma.

In fact, I should try to find what causes a full moon, to get them to appear everytime.

**Comment by AgentRev** [comment]
According to an Armaholic post, full moons appear in Arma 2 every 6th of the month, so maybe setting the date to July 6th will do.

Otherwise, in real life, there is supposed to be a full moon on June 20, 2035. If that doesn't work, I'll just find one manually.

**Comment by AgentRev** [comment]
Best is Jun 10, 2035, since the moon is directly opposite of the sun, and so shines all night.

**Comment by AgentRev** [comment]
You should try to contact His_Shadow on his own forum, because nearly nobody reads their PMs on the BIS forums, since all notification options are disabled by default.

I had like 4 PMs in my inbox that I received weeks earlier, and I didn't even know that I got them because I never received any mail and never got a popup on the forum.

It says his last activity was on Jul 31, so he didn't read it.

**Comment by JoSchaap** [comment]
Will do that.

Also i'm going to gamescon on sunday the 25th
Just to meet matt lightfoot and the arma3 devs in person.
And to play dayz standalone and the arma 3 pre release version :)

**Comment by AgentRev** [comment]
Lucky you :stuck_out_tongue: only a 3h drive
